#! /usr/bin/env python
PACKAGE='simple_image_processing'
from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("blue_thres",  int_t, 0, "Threshold",   80, 0, 255)
gen.add("erode_size",  int_t, 0, "Erode Size",  7, 1, 21)
gen.add("dilate_size",  int_t, 0, "Dilate Size", 3, 1, 21)
gen.add("hough_rho_res",  int_t, 0, "Rho resolution (pix)", 10, 1, 500)
gen.add("hough_theta_res",  double_t, 0, "Theta resolution (rad)", 0.05, 0.01, 1.0)
gen.add("hough_threshold",  int_t, 0, "Accumulator threshold", 25, 10, 200)
gen.add("hough_min_length",  int_t, 0, "Minimum line length (pix)", 250, 5, 500)
gen.add("hough_max_gap",  int_t, 0, "Max gap betweeen line segments (pix)", 50, 5, 500)



gen.add("lwr_ylw_sclr_g",  int_t, 0, "Lower threshold for yellow lane lines G component", 0, 0, 255)
gen.add("lwr_ylw_sclr_r",  int_t, 0, "Lower threshold for yellow lane lines R component", 0, 0, 255)

gen.add("hgr_ylw_sclr_b", int_t, 0 , "Higher threshold for yellow lane lines B component", 90, 0, 255)
gen.add("hgr_ylw_sclr_g", int_t, 0 , "Higher threshold for yellow lane lines G component", 254, 0, 255)
gen.add("hgr_ylw_sclr_r", int_t, 0 , "Higher threshold for yellow lane lines R component", 254, 0, 255)

gen.add("gray_thresh_low",  int_t, 0, "Gray threshold in inRange thresholding", 0, 0, 255)
gen.add("gray_thresh_high",  int_t, 0, "Gray threshold in inRange thresholding", 254, 0, 255)

gen.add("lowThreshVal",  int_t, 0, "Gray threshold after gaussian and before canny", 50, 0, 255)
gen.add("targetThreshVal",  int_t, 0, "Gray threshold after gaussian and before canny", 254, 0, 255)

gen.add("canny_thresh_low",  int_t, 0, "Canny threshold low", 40, 0, 255)
gen.add("canny_thresh_high",  int_t, 0, "Canny threshold low", 160, 0, 255)


exit(gen.generate(PACKAGE, PACKAGE, "RgbHough"))
